testgres/pubsub.py

```diff
@@ -0,0 +1,200 @@
+>>> with get_new_node() as nodeA, get_new_node() as nodeB:
+...     nodeA.init(allow_logical=True).start()
+...     nodeB.init().start()
+...
+...     # create same table both on publisher and subscriber
+...     create_table = 'create table test (a int, b int)'
+...     nodeA.safe_psql(create_table)
+...     nodeB.safe_psql(create_table)
+...
+...     # create publication
+...     pub = nodeA.publish('mypub')
+...     # create subscription
+...     sub = nodeB.subscribe(pub, 'mysub')
+...
+...     # insert some data to the publisher's node
+...     nodeA.execute('insert into test values (1, 1), (2, 2)')
+...
+...     # wait until changes apply on subscriber and check them
+...     sub.catchup()
+...
+...     # read the data from subscriber's node
+...     nodeB.execute('select * from test')
+PostgresNode(name='...', port=..., base_dir='...')
+PostgresNode(name='...', port=..., base_dir='...')
+''
+''
+[(1, 1), (2, 2)]
+"""
+
+from six import raise_from
+
+from .defaults import default_dbname, default_username
+from .exceptions import CatchUpException
+from .utils import options_string
+
+
+class Publication(object):
+    def __init__(self, name, node, tables=None, dbname=None, username=None):
+        """
+        Constructor. Use :meth:`.PostgresNode.publish()` instead of direct
+        constructing publication objects.
+
+        Args:
+            name: publication name
+            node: publisher's node
+            tables: tables list or None for all tables
+            dbname: database name used to connect and perform subscription
+            username: username used to connect to the database
+        """
+        self.name = name
+        self.node = node
+        self.dbname = dbname or default_dbname()
+        self.username = username or default_username()
+
+        # create publication in database
+        t = "table " + ", ".join(tables) if tables else "all tables"
+        query = "create publication {} for {}"
+        node.safe_psql(query.format(name, t), dbname=dbname, username=username)
+
+    def drop(self, dbname=None, username=None):
+        """
+        Drop publication
+        """
+        self.node.safe_psql(
+            "drop publication {}".format(self.name),
+            dbname=dbname,
+            username=username)
+
+    def add_tables(self, tables, dbname=None, username=None):
+        """
+        Add tables to the publication. Cannot be used if publication was
+        created with empty tables list.
+
+        Args:
+            tables: a list of tables to be added to the publication
+        """
+        if not tables:
+            raise ValueError("Tables list is empty")
+
+        query = "alter publication {} add table {}"
+        self.node.safe_psql(
+            query.format(self.name, ", ".join(tables)),
+            dbname=dbname or self.dbname,
+            username=username or self.username)
+
+
+class Subscription(object):
+    def __init__(self,
+                 name,
```

Comment:
It's better to place `name` right after the mandatory args. Chances are we'll come up with some name generation facility, but we won't be able to make `name` optional without moving it, which is something I'd like to avoid.